src/libraries/System.Net.WebProxy/tests/WebProxyTest.cs

```diff
@@ -16,7 +16,12 @@ public static IEnumerable<object[]> Ctor_ExpectedPropertyValues_MemberData()
             yield return new object[] { new WebProxy(), null, false, false, Array.Empty<string>(), null };
 
             yield return new object[] { new WebProxy("http://anything"), new Uri("http://anything"), false, false, Array.Empty<string>(), null };
+            yield return new object[] { new WebProxy("http://anything:42"), new Uri("http://anything:42"), false, false, Array.Empty<string>(), null };
+            yield return new object[] { new WebProxy("anything:42"), new Uri("http://anything:42"), false, false, Array.Empty<string>(), null };
             yield return new object[] { new WebProxy("anything", 42), new Uri("http://anything:42"), false, false, Array.Empty<string>(), null };
+            yield return new object[] { new WebProxy("http://anything", 42), new Uri("http://anything:42"), false, false, Array.Empty<string>(), null };
+            yield return new object[] { new WebProxy("http://anything:123", 42), new Uri("http://anything:42"), false, false, Array.Empty<string>(), null };
```

Comment:
I wonder if we should rather throw in this case when there are 2 ports defined by the user - in the address and then explicitly in the `WebProxy` arg. That would be likely a breaking change though :(.

Reply:
Today you would get `"http://http://anything:123:42"` so this may already be an improvement 😄

The port being overwritten here seems like reasonable behavior to me, not sure it's worth adding an exception message.